### PR TITLE
[WIP DO NOT REVIEW] chore(content-server): clean up backbone code for simple routes

### PR DESCRIPTION
**Pull request by millsoper**
# WIP DO NOT REVIEW

Because:

* Now that we've rolled out the simple routes to 100%, we want to clean up the remaining backbone code.

This commit:

* Removes all mustache files associated with the simple routes (cannot_create_account, clear_storage, cookies_disabled, legal, and tos)

* Removes all view js files (cannot_create_account, clear_storage, cookies_disabled, tos)

* Removes all test spec files (cannot_create_account, clear_storage, cookies_disabled, tos)

* Removes references to the Backbone views from the router file

Closes #FXA-7398

## Because

-

## This pull request

-

## Issue that this pull request solves

Closes: # (issue number)

## Checklist

_Put an `x` in the boxes that apply_

- [ ] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-content-server/app/scripts/lib/router.js [hunk 1]
```diff
@@ -487,6 +477,21 @@ Router = Router.extend({
     return false;
   },
 
+  createReactViewHandler(routeName, Path, additionalParams) {
```
Comment (pdehaan) [inline packages/fxa-content-server/app/scripts/lib/router.js:480]: Is `Path` used? I can't see it.
I do see one instance of `this.createReactViewHandler('cookies_disabled', {…})` above on L176 that seems to be passing some additional params (apart from `routeName`) so I was wondering if those are just getting dropped since they might be getting passed as `Path` and not `additionalParams` argument.

Comment (millsoper) [reply packages/fxa-content-server/app/scripts/lib/router.js:480]: Oh that's a good catch -- yeah I was trying to mirror the pre-existing method but I think you're right, this would be an improvement. I will update!

File: packages/fxa-content-server/grunttasks/remarkable.js [hunk 2]
```diff
@@ -2,48 +2,13 @@
...
-  function rename(destPath, destFile) {
-    // Normalize the filenames to use the locale name.
-    // add the extension here, instead of using grunt-remarkable's
-    // extension generator, to get locale names in uppercase
-
-    const lang = i18n.localeFrom(destFile.split('/')[0]);
-    return path.join(destPath, lang + '.html');
-  }
-
   grunt.config('remarkable', {
```
Comment (LZoog) [inline packages/fxa-content-server/grunttasks/remarkable.js:6]: Are we still using this for something else or can this file be removed?

File: packages/fxa-content-server/tests/functional/pages.js [hunk 3]
```diff
@@ -7,35 +7,21 @@
 const { registerSuite } = intern.getInterface('object');
 var url = intern._config.fxaContentRoot;
 
-// Commented out pages here likely mean the React version is rolled out to 100%
-// and we don't want to check these for `#stage header` selectors.
```
Comment (LZoog) [inline packages/fxa-content-server/tests/functional/pages.js:11]: Might leave this comment as-is since we will be likely commenting out other tests here as we roll these out to 100% but prior to removing this code.

File: packages/fxa-content-server/app/scripts/lib/router.js [hunk 4]
```diff
@@ -487,6 +477,21 @@ Router = Router.extend({
     return false;
   },
 
+  createReactViewHandler(routeName, additionalParams) {
```
Comment (LZoog) [inline packages/fxa-content-server/app/scripts/lib/router.js:480]: Can we call this function [here and remove the duplicate lines](https://github.com/mozilla/fxa/blob/a74024c1fc4eff5c329f15a0560dd97a4af30249/packages/fxa-content-server/app/scripts/lib/router.js#L504)?

File: packages/fxa-content-server/app/scripts/lib/router.js [hunk 5]
```diff
@@ -179,36 +173,32 @@ Router = Router.extend({
...
+      });
     },
     'force_auth(/)': createViewHandler(ForceAuthView),
     'inline_totp_setup(/)': createViewHandler(InlineTotpSetupView),
     'inline_recovery_setup(/)': createViewHandler(InlineRecoverySetupView),
     'legal(/)': function () {
-      this.createReactOrBackboneViewHandler('legal', 'legal');
+      this.createReactViewHandler('legal');
     },
     'legal/privacy(/)': createViewHandler('pp'),
```
Comment (LZoog) [inline packages/fxa-content-server/app/scripts/lib/router.js:199]: Ack, I don't know why this is showing `createViewHandler` for `legal/privacy`, I feel like even QA verified the React version was showing here? 🥴 I think you need to change this one too to `this.createReactViewHandler`. We may want QA to look at that page with the next tag too in case this page wasn't tested somehow before...

File: packages/fxa-content-server/grunttasks/remarkable.js (hunk 2 above)
Comment (pdehaan) [reply packages/fxa-content-server/grunttasks/remarkable.js:6]: As far as I can tell, we only used remarkable for TOS/PP stuffs, so we might be able to do some additional cleanup.

```sh
git grep -n "remarkable" packages | cat

packages/fxa-content-server/.nsprc:4:  "comment_745": "745 is RegExp denial by grunt-remarkable.",
packages/fxa-content-server/grunttasks/l10n-generate-tos-pp.js:14:        'remarkable:tos_pp',
packages/fxa-content-server/grunttasks/remarkable.js:6:  grunt.config('remarkable', {
packages/fxa-content-server/grunttasks/replace.js:21:          // such as `    [`, this creates a <code> block when parsed by remarkable
packages/fxa-content-server/package.json:101:    "grunt-remarkable": "1.1.0",
```

if https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/grunttasks/l10n-generate-tos-pp.js isn't used anymore, we might be able to start recursively removing more scripts and tasks:

https://github.com/mozilla/fxa/blob/658ddb7b31c237d397a201b10871b2c309bcef3e/packages/fxa-content-server/grunttasks/l10n-generate-tos-pp.js#L7-L18

File: packages/fxa-content-server/app/scripts/lib/router.js (hunk 5 above)
Comment (millsoper) [reply packages/fxa-content-server/app/scripts/lib/router.js:199]: 😅 ack!

File: packages/fxa-content-server/app/scripts/lib/router.js (hunk 4 above)
Comment (millsoper) [reply packages/fxa-content-server/app/scripts/lib/router.js:480]: SMART! 🧠

File: packages/123done/static/pp.txt [hunk 6]
```diff
@@ -1,5 +1,5 @@
 This is a Privacy Policy document for 123done.org.  If this were a
-real site, it would give you the policies which we adhere to as we 
+real site, it would give you the policies which we adhere to as we
```
Comment (LZoog) [inline packages/123done/static/pp.txt:2]: Maybe revert this change so this file is out of the diff?

File: packages/fxa-content-server/.nsprc [hunk 7]
```diff
@@ -26,5 +26,5 @@
     "https://npmjs.com/advisories/1426",
     "https://npmjs.com/advisories/1488",
     "https://npmjs.com/advisories/1500"
-   ]
```
Comment (LZoog) [inline packages/fxa-content-server/.nsprc:29]: Maybe revert this change so this file is out of the diff?

File: packages/fxa-content-server/app/scripts/lib/router.js [hunk 8]
```diff
@@ -179,36 +173,34 @@ Router = Router.extend({
...
     },
     'force_auth(/)': createViewHandler(ForceAuthView),
     'inline_totp_setup(/)': createViewHandler(InlineTotpSetupView),
     'inline_recovery_setup(/)': createViewHandler(InlineRecoverySetupView),
     'legal(/)': function () {
-      this.createReactOrBackboneViewHandler('legal', 'legal');
+      this.createReactViewHandler('legal');
+    },
+    'legal/privacy(/)': function () {
+      this.createReactViewHandler('pp');
```
Comment (LZoog) [inline packages/fxa-content-server/app/scripts/lib/router.js:200]: Should this be `this.createReactViewHandler('legal/privacy');`?

File: packages/fxa-content-server/app/tests/spec/lib/router.js [hunk 9]
```diff
@@ -596,7 +595,7 @@ describe('lib/router', () => {
         sinon.stub(router, 'isInReactExperiment').callsFake(() => true);
         routeHandler = router.createReactOrBackboneViewHandler(
           'cannot_create_account',
```
Comment (LZoog) [inline packages/fxa-content-server/app/tests/spec/lib/router.js:597]: I think we will also want to change `'cannot_create_account'` to its view equivalent; passing in `SignInPasswordView` is funky when leaving `cannot_create_account` as-is.

File: packages/fxa-content-server/tests/functional/pages.js [hunk 10]
```diff
@@ -13,29 +13,14 @@ var pages = [
   '',
   'authorization',
   'boom',
-  // 'cannot_create_account',
```
Comment (LZoog) [inline packages/fxa-content-server/tests/functional/pages.js:16]: Sorry if my comment [here](https://github.com/mozilla/fxa/pull/15340/#discussion_r1201321288) was unclear. We should remove the commented out tests, I just think we should leave this comment above for when we comment out tests in the future:

> // Commented out pages here likely mean the React version is rolled out to 100%
> // and we don't want to check these for `#stage header` selectors.

**Review by LZoog — COMMENTED** [review]
So much code removal 😍

I'll look into the CI error now, just wanted to take a quick look at the changes first.

**Comment by LZoog** [comment]
I cherry-picked in the CI fix to see if this passes with it to avoid fixing merge conflicts for now. Will have to circle back to this PR to review/make changes, rebase, and push.

**Comment by LZoog** [comment]
closing in favor of #15561
